modules/alloxBidAdapter.js

```diff
@@ -0,0 +1,239 @@
+          .reduce((pre, cur) => `${pre}${createTrackPixelHtml(cur.url)}`, '');
+        bid.ad += impTrackersHtml;
+      };
+    });
+    return response;
+  }
+});
+
+export const storage = getStorageManager({ bidderCode: BIDDER_CODE });
+
+export const spec = {
+
+  code: BIDDER_CODE,
+  aliases: ['alx'],
+
+  isBidRequestValid: function(bidRequest) {
+    const alloxIds = getAlloxIds();
+    bidRequest.params.nidanId = isEmpty(alloxIds.nidanId) ? bidRequest.params.nidanId || storage.getDataFromLocalStorage(NIDAN_ID_KEY) : alloxIds.nidanId;
+    bidRequest.params.daisyId = isEmpty(alloxIds.daisyId) ? bidRequest.params.daisyId : alloxIds.daisyId;
+    return !!bidRequest.params.placementId && includeNidanIdOrDaisyID(bidRequest.params);
+  },
+
+  buildRequests: function(validBidRequests, bidderRequest) {
+    return validBidRequests.map(validBidRequest => {
+      const context = { mediaType: getMediaType(validBidRequest) };
+      const ortbRequest = converter.toORTB({
+        bidRequests: [validBidRequest],
+        bidderRequest: bidderRequest,
+        context: context
+      });
+
+      const serverRequest = {
+        p: validBidRequest.params.placementId,
+        t: `web_${generateUUID()}-${validBidRequest.bidId}`,
+        bw: window.innerWidth,
+        impid: validBidRequest.bidId,
+        n: validBidRequest.params.nidanId,
+        d: validBidRequest.params.daisyId
+      };
+
+      setRequestParamFromURLParameter(serverRequest, 'debug', QUERY_REG.DEBUG);
+      setRequestParamFromURLParameter(serverRequest, 'stb', QUERY_REG.STB);
+      setRequestParamFromURLParameter(serverRequest, 'upt', QUERY_REG.TARGET);
+      setRequestParamFromURLParameter(serverRequest, 'test', QUERY_REG.TEST);
+
+      const request = {
+        method: 'POST',
+        url: ENDPOINT_URL,
+        data: deepClone(serverRequest),
+        options: {
+          withCredentials: false
+        },
+        ortbRequest,
+        validBidRequest
+      };
+      return request;
+    });
+  },
+
+  interpretResponse: function(serverResponse, request) {
+    if (isEmpty(serverResponse.body)) return [];
+
+    const bids = converter.fromORTB({
+      response: serverResponse.body,
+      request: request.ortbRequest
+    }).bids;
+
+    if (storage.localStorageIsEnabled()) {
+      const trackersValue = storage.getDataFromLocalStorage(STORAGE_KEY);
+      const trackers = trackersValue ? JSON.parse(trackersValue) : {};
+      const now = new Date().getTime();
+      Object.keys(trackers).forEach(k => {
+        if (now - trackers[k].date > STORAGE_TIMEOUT) {
+          delete trackers[k];
+        }
+      })
+      const trackersObj = isEmpty(serverResponse.body.seatbid) ? { trackers: deepAccess(serverResponse, 'body.ext.trackers') } : bids[0];
+      trackers[request.validBidRequest.adUnitId] = extractTrackers(trackersObj);
+      storage.setDataInLocalStorage(STORAGE_KEY, JSON.stringify(trackers));
```

Comment:
we should consider putting the JSON.parse and the stringify call into local storage into a try/catch